### PR TITLE
Enable event listeners, event handlers and organization switch grant 

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

The B2B organization management feature will be available as a new feature from the WSO2 Identity server. Hence the following supportive event handlers and grant types should be enabled by default.

### Related Issues
- https://github.com/wso2/product-is/issues/16359

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 1]
```diff
@@ -478,6 +478,14 @@
...
             </SupportedGrantType>
             {% endif %}
+            {% if oauth.grant_type.organization_switch.enable is sameas true %}
+            <SupportedGrantType>
+                <GrantTypeName>organization_switch</GrantTypeName>
+                <GrantTypeHandlerImplClass>{{oauth.grant_type.organization_switch.grant_handler}}</GrantTypeHandlerImplClass>
+                <GrantTypeValidatorImplClass>{{oauth.grant_type.organization_switch.grant_validator}}</GrantTypeValidatorImplClass>
+                <IdTokenAllowed>true</IdTokenAllowed>
+            </SupportedGrantType>
+            {% endif %}
```
Comment (sadilchamishka) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:488]: [[oauth.custom_grant_type]]
name="organization_switch"
grant_handler="org.wso2.carbon.identity.oauth2.grant.organizationswitch.OrganizationSwitchGrant"
grant_validator="org.wso2.carbon.identity.oauth2.grant.organizationswitch.OrganizationSwitchGrantValidator"
[oauth.custom_grant_type.properties]
IdTokenAllowed=true

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -1741,6 +1749,18 @@
...
+                               type="org.wso2.carbon.identity.application.mgt.listener.ApplicationMgtListener"
+                               name="org.wso2.carbon.identity.organization.management.application.listener.FragmentApplicationMgtListener"
+                               orderId="{{event.default_listener.fragment_app_mgt_listener.priority}}"
+                               enable="{{event.default_listener.fragment_app_mgt_listener.enable}}"/>
+
+        <EventListener id="organization_user_operation_listener"
+                                       type="org.wso2.carbon.user.core.listener.UserOperationEventListener"
+                                       name="org.wso2.carbon.identity.organization.management.role.management.service.listener.OrganizationUserOperationEventListener"
+                                       orderId="{{event.default_listener.organization_user_operation_listener.priority}}"
+                                       enable="{{event.default_listener.organization_user_operation_listener.enable}}"/>
```
Comment (sadilchamishka) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1763]: Corresponding deployment.toml configurations which was used previously.


```
[[event_listener]]
id="fragment_app_mgt_listener"
type="org.wso2.carbon.identity.application.mgt.listener.ApplicationMgtListener"
name="org.wso2.carbon.identity.organization.management.application.listener.FragmentApplicationMgtListener"
order="50"
enable="true"

[[event_listener]]
id="organization_user_operation_listener"
type="org.wso2.carbon.user.core.listener.UserOperationEventListener"
name="org.wso2.carbon.identity.organization.management.role.management.service.listener.OrganizationUserOperationEventListener"
order="115"
enable="true"
```

File: features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json [hunk 3]
```diff
@@ -227,5 +227,13 @@
...
   "identity_mgt.events.schemes.WorkflowPendingUserAuthnHandler.subscriptions": [
     "PRE_AUTHENTICATION"
+  ],
+  "identity_mgt.events.schemes.OrganizationCreationHandler.module_index": "36",
+  "identity_mgt.events.schemes.OrganizationCreationHandler.subscriptions": [
+    "POST_ADD_ORGANIZATION"
+  ],
+  "identity_mgt.events.schemes.GovernanceConfigUpdateHandler.module_index": "37",
+  "identity_mgt.events.schemes.GovernanceConfigUpdateHandler.subscriptions": [
+    "POST_ADD_ORGANIZATION"
```
Comment (sadilchamishka) [inline features/identity-event/org.wso2.carbon.identity.event.server.feature/resources/org.wso2.carbon.identity.event.server.feature.default.json:237]: Corresponding deployment.toml configurations which was used previously.

```
[[event_handler]]
name="OrganizationCreationHandler"
subscriptions=["POST_ADD_ORGANIZATION"]

[[event_handler]]
name= "GovernanceConfigUpdateHandler"
subscriptions =["POST_ADD_ORGANIZATION"]
```

**Review by AnuradhaSK — APPROVED** [review]

**Comment by sadilchamishka** [comment]
This PR cause integration test failures. Hence we can merge without PR builder. 
The test failure is fixed by the PR - https://github.com/wso2/product-is/pull/16402

All the integration tests were executed locally as shown in below SS.
Screenshot 2023-08-08 at 08 54 06

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 4]
```diff
@@ -1749,6 +1757,18 @@
...
```
Comment (AnuradhaSK) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:1771]: Shall we align all the params against the `id` parameter? (Fix the indentation)

Comment (sadilchamishka) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed by - https://github.com/wso2/carbon-identity-framework/pull/4825/commits/8433f50d175e8247aa74cb121e4bf2dc06f7e4d8

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**